File: Login_Registrationandhomepagelinked/app/src/main/java/com/sp/mad_project/MainActivity.java
package com.sp.mad_project;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Reference the buttons
        Button buttonToReport = findViewById(R.id.report_button);
        Button post = findViewById(R.id.camera_icon);
        Button search = findViewById(R.id.search_button);
        Button chat = findViewById(R.id.chat_button);

        // Set click listeners for the buttons
        post.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Navigate to posts activity
                Intent intent = new Intent(MainActivity.this, posts.class);
                startActivity(intent);
            }
        });

        search.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Navigate to Search activity
                Intent intent = new Intent(MainActivity.this, Search.class);
                startActivity(intent);
            }
        });

        chat.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Navigate to Search activity
                Intent intent = new Intent(MainActivity.this, userSelectScreen.class);
                startActivity(intent);
            }
        });

        buttonToReport.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Navigate to ReportActivity
                Intent intent = new Intent(MainActivity.this, ReportActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Login_Registrationandhomepagelinked/app/src/main/java/com/sp/mad_project/MainActivity.kt
package com.sp.mad_project

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Reference the button
        val buttonToReport = findViewById<Button>(R.id.report_button)
        val post = findViewById<Button>(R.id.camera_icon)
        val search = findViewById<Button>(R.id.search_button)
        // Set a click listener for the button

        post.setOnClickListener { v: View? ->
            // Navigate to ReportActivity
            val intent = Intent(this@MainActivity, posts::class.java)
            startActivity(intent)
        }

        search.setOnClickListener { v: View? ->
            // Navigate to ReportActivity
            val intent = Intent(this@MainActivity, Search::class.java)
            startActivity(intent)
        }


        buttonToReport.setOnClickListener { v: View? ->
            // Navigate to ReportActivity
            val intent = Intent(this@MainActivity, ReportActivity::class.java)
            startActivity(intent)
        }
    }
}